src/Core/Application/RealTime/UseCase/FindServiceCategory/FindServiceCategory.php

```diff
@@ -60,11 +79,44 @@ public function __invoke(FindServiceCategoryPresenterInterface $presenter): void
     }
 
     /**
-     * @param ServiceCategory[] $serviceCategories
+     * @param Tag[] $serviceCategories
      * @return FindServiceCategoryResponse
      */
     private function createResponse(array $serviceCategories): FindServiceCategoryResponse
     {
         return new FindServiceCategoryResponse($serviceCategories);
     }
+
+    /**
+     * @param FindServiceCategoryPresenterInterface $presenter
+     * @return void
+     */
+    private function handleImcompatibleBBDOVersion(FindServiceCategoryPresenterInterface $presenter): void
+    {
+        $message = 'BBDO protocol version enabled not compatible with this feature. Version needed '
+            . self::MINIMUM_BBDO_VERSION_SUPPORTED . ' or higher';
+        $this->error($message);
+        $presenter->setResponseStatus(new ImcompatibilityResponse($message));
+    }
+
+    /**
+     * Checks if at least on monitoring server has BBDO protocol in version 3.0.0
+     *
+     * @return boolean
+     */
+    private function isBBDOVersionCompatible(): bool
```

Comment:
If you want you can use a trait to share this part of code with the other use case **FindHostCategory**